src/Mutex/Impl/Internal/MutexWorkflow.workflow.cs

```diff
@@ -0,0 +1,55 @@
+using Temporalio.Workflows;
+
+namespace TemporalioSamples.Mutex.Impl.Internal;
+
+[Workflow]
+internal class MutexWorkflow
+{
+    private readonly ILockHandler lockHandler = WorkflowMutex.CreateLockHandler();
+    private readonly Queue<LockRequest> requests = new();
+
+    [WorkflowRun]
+    public async Task RunAsync(MutexWorkflowInput? input = null)
```

Comment:
Don't have to do it, but there's a neat feature in .NET SDK not in other SDKs - you can have a constructor with `[WorkflowInit]` that accepts this input there

Reply:
I understood that the constructor is called with an input, and you can populate the Queue there, but RunAsync still needs to have an argument, which is not used if there is a constructor. It looks strange in the end

Reply:
Concur, but it should be noted that "signal with start" in Temporal, for special ordering reasons, will call a signal handler _before_ "run" is called. But in your case you only ever send signal-with-start with no initial requests, so you're good I think.